src/XComponent.Functions.Test/FunctionsManagerTest.cs

```diff
@@ -168,14 +169,54 @@ public interface ISender
                 PublicMember = publicMemberAfter,
             };
 
+            var sender = new object(); 
+            var context = new object();
             var functionsManager = new FunctionsManager("component", "statemachine");
+            functionsManager.AddTask(null, null, null, context, sender);
             functionsManager.ApplyFunctionResult(functionResult,
                     publicMemberBefore,
                     null,
-                    new object(),
-                    new object());
+                    context,
+                    sender);
 
             Assert.AreEqual("after", publicMemberBefore.State);
         }
+
+        [Test]
+        public void ApplyFunctionResultShouldThrowSerializationException()
+        {
+            var publicMemberBefore = new PublicMember() { State = "before" };
+
+            var publicMemberAfter = new JObject();
+            publicMemberAfter.Add("State", new JValue("after"));
+
+            var functionResult = new FunctionResult()
+            {
+                PublicMember = "Hello",
+            };
+
+            var functionsManager = new FunctionsManager("component", "statemachine");
+            
+            Assert.Throws<SerializationException>(() => functionsManager.ApplyFunctionResult(functionResult,
+                publicMemberBefore,
+                null,
+                new object(),
+                new object()));
+        }
+
+        [Test]
+        public void ApplyFunctionResultShouldThrowSenderNotFoundException()
+        {
+            var publicMemberAfter = new JObject();
+            publicMemberAfter.Add("State", new JValue("after"));
+
+            var functionsManager = new FunctionsManager("component", "statemachine");
+
+            Assert.Throws<Exception>(() => functionsManager.ApplyFunctionResult(null,
```

Comment:
write a case where you have a sender call in the result.

looking at your test I don't know why it should throw.

Reply:
the case where there's no sender call is not an error!
you shouldn't throw in this case!!!